ivadomed/testing.py

```diff
@@ -221,8 +227,12 @@ def run_inference(test_loader, model, model_params, testing_params, ofolder, cud
                                                         fname_out=fname_pred,
                                                         slice_axis=slice_axis,
                                                         kernel_dim='3d',
-                                                        bin_thr=0.5 if testing_params["binarize_prediction"] else -1)
-                    preds_npy_list.append(output_nii.get_fdata().transpose(3, 0, 1, 2))
+                                                        bin_thr=bin_thr)
+                    output_data = output_nii.get_fdata().transpose(3, 0, 1, 2)
+                    if testing_params["binarize_predictions"] >= 0:
+                        output_data = threshold_predictions(output_data, thr=testing_params["binarize_predictions"])
```

Comment:
The thresholding with `binarize_predicitons` should also be in the `evaluation.py`. The command `eval` uses directly the output in the pred_masks folder to compute the metrics, but with `binarize_niftis` we don't necessarily want to have the metrics from the generated masks. This would be another argument in favor of having only one param (only `binarize_predictions` in the config file :}

Reply:
updated evaluation.py here 7aa3e21

Reply:
As we decided to have only one param... I can undo this, right? Since the predictions will always be generated according to testing_params. If binarise_prediction = -1 --> metrics_testing, niftis and metrics_eval are soft, otherwise they are hard, all with the same threshold. Are we on the same page? :-)

Reply:
Yes I totally agree :D!